src/opaque.rs

```diff
@@ -658,11 +629,7 @@ impl<CS: CipherSuite> ClientLogin<CS> {
     }
 }
 
-type ClientLoginFinishResult = (
-    LoginThirdMessage,
-    Vec<u8>,
-    GenericArray<u8, <Sha256 as Digest>::OutputSize>,
-);
+type ClientLoginFinishResult = (LoginThirdMessage, Vec<u8>, GenericArray<u8, ExportKeySize>);
```

Comment:
Not that this is necessarily part of your PR, but I'm repetitively struggling at remembering the nature of the components of this tuple, something a proper struct with labels (third_message, shared_secret, export-key) would help with.

Reply:
Still there, but it's a nit

Reply:
Since this results in a change to the API, I will leave it to a future discussion where we consider the readability of the user-facing API as a whole